src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -0,0 +1,320 @@
+                                                                                      s_localizableTitle,
+                                                                                      s_staticAbstractMessage,
+                                                                                      DiagnosticCategory.Usage,
+                                                                                      DiagnosticSeverity.Error,
+                                                                                      isEnabledByDefault: true,
+                                                                                      s_localizableDescription,
+                                                                                      helpLinkUri: @" https://aka.ms/dotnet-warnings/preview-features");
+        internal static DiagnosticDescriptor ImplementsEmptyPreviewInterfaceRule = new(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_implementsEmptyPreviewInterface,
+                                                                                      DiagnosticCategory.Usage,
+                                                                                      DiagnosticSeverity.Error,
+                                                                                      isEnabledByDefault: true,
+                                                                                      s_localizableDescription,
+                                                                                      helpLinkUri: @" https://aka.ms/dotnet-warnings/preview-features");
+
+#pragma warning restore RS0030
+        private const string RequiresPreviewFeaturesAttribute = nameof(RequiresPreviewFeaturesAttribute);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(GeneralPreviewFeatureAttributeRule, StaticAbstractIsPreviewFeatureRule, ImplementsEmptyPreviewInterfaceRule);
+
+        private enum PreviewFeatureUsageType
+        {
+            General = 0,
+            StaticAbstract,
+            EmptyInterface
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(context =>
+            {
+                Compilation compilation = context.Compilation;
+                IAssemblySymbol? thisAssembly = compilation.Assembly;
+                ImmutableArray<AttributeData> attributes = thisAssembly.GetAttributes();
+
+                AttributeData? attribute = attributes.FirstOrDefault(x => x.AttributeClass.Name == RequiresPreviewFeaturesAttribute);
```

Comment:
I can't do a symbol comparison here yet (already called this out in the PR description). The RequiresPreviewFeaturesAttribute was introduced in .NET6P5, and this project doesn't use that version yet. I'd like to fix this code to use symbol comparison in a later PR after:
1. The P6 sdk becomes public
2. I update the 6.0.0.x branch to use the P6 sdk

Reply:
@pgovind I think symbol comparison can be done without .NET 6 P5.

If the blocker is unit tests, then the attribute in tests just needs to be defined in the correct namespace.

Reply:
Since this is used only for null check. This could use `Any` instead, or it can use HasAttribute whenever symbol comparison is used.

Reply:
Sounds better to add mock attribute class for the unit testing and do symbol comparison here, it could fail fast if the new attribute is not found

Reply:
Fixed now